
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />

    <title>Welcome to QTL_BSA-Sorghum’s documentation! &#8212; QTL_BSA-Sorghum 1.0 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/alabaster.css" />
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="QTLSorghum" href="qtl.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <section id="welcome-to-qtl-bsa-sorghum-s-documentation">
<h1>Welcome to QTL_BSA-Sorghum’s documentation!<a class="headerlink" href="#welcome-to-qtl-bsa-sorghum-s-documentation" title="Permalink to this headline">¶</a></h1>
<div class="toctree-wrapper compound">
</div>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="qtl.html">QTLSorghum</a><ul>
<li class="toctree-l2"><a class="reference internal" href="qtl.html#for-more-detailed-instructions-please-read-the-vignette-here"><strong>For more detailed instructions please read the vignette here</strong></a></li>
<li class="toctree-l2"><a class="reference internal" href="qtl.html#for-updates-read-the-news-md">For updates read the NEWS.md</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#installation">Installation</a><ul>
<li class="toctree-l2"><a class="reference internal" href="qtl.html#abstract">Abstract</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#examples">Examples:</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#load-install-libraries">Load/install libraries</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#set-the-working-directory-to-where-vcf-file-is-stored-in-file-system">Set the Working Directory to where VCF file is stored in file system</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#vcf-file-must-only-contain-bialleleic-variants-filter-upstream-e-g-with-bcftools-view-m2-m2-also-the-qtlseqr-functions-will-only-take-snps-ie-length-of-ref-and-alt-1">Vcf file must only contain bialleleic variants. (filter upstream, e.g., with bcftools view -m2 -M2), also the QTLseqR functions will only take SNPS, ie, length of REF and ALT== 1</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#call-the-parser">Call the Parser</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#preview-the-csv-file">Preview the CSV file</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#invoke-unique-command-to-extract-sample-names-reverse-comapatible-to-the-vcf">Invoke unique command to extract Sample names reverse comapatible to the VCF</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#inspect-the-head-of-the-df-object">Inspect the head of the df object</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#g-distribution-plot">G’ Distribution Plot</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#plot-g-statistic-distribution">Plot G Statistic Distribution</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#use-rmvp-package-to-view-snps-on-chromosomes-contigs">Use RMVP package to view SNPs on chromosomes/contigs</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#export-summary-csv">Export summary CSV</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#preview-the-summary-qtl">Preview the Summary QTL</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#p2-p1-qtl-is-present">p2 &gt;&gt; p1 QTL is present</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#however-ns-c-1-is-not-true">However, ns &gt;&gt; C &gt;&gt; 1 is NOT TRUE</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#standard-parser-output-makes-a-list-of-chromosome-as-named-in-vcf-file">Standard Parser Output makes a list of Chromosome as named in VCF file</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#also-find-unique-sample-names-reverse-compatible-with-vcf-file">Also find unique sample names reverse compatible with VCF file</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#id1">Preview the CSV file</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#plot-histograms-associated-with-filtering-arguments-to-determine-if-cut-off-values-are-appropriate">Plot histograms associated with filtering arguments to determine if cut off values are appropriate</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#id2">Plot G Statistic Distribution</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#lets-take-a-look-at-snps-per-chromosome-using-rmvp-package">Lets take a look at SNPs per chromosome using rMVP Package</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#id3">Export summary CSV</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#preview-the-qtl-summary">Preview the QTL Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#looks-dense">Looks Dense</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#filter-low-allelic-depth-frequencies">Filter Low Allelic Depth Frequencies</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#preview-the-plot-with-idenitfied-snp-positions">Preview the plot with idenitfied SNP positions</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#investigate-snp-pos-24525659">Investigate SNP&#64;POS 24525659</a></li>
</ul>
</div>
</section>
<section id="indices-and-tables">
<h1>Indices and tables<a class="headerlink" href="#indices-and-tables" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li><p><a class="reference internal" href="genindex.html"><span class="std std-ref">Index</span></a></p></li>
<li><p><a class="reference internal" href="py-modindex.html"><span class="std std-ref">Module Index</span></a></p></li>
<li><p><a class="reference internal" href="search.html"><span class="std std-ref">Search Page</span></a></p></li>
</ul>
</section>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="#">QTL_BSA-Sorghum</a></h1>








<h3>Navigation</h3>
<ul>
<li class="toctree-l1"><a class="reference internal" href="qtl.html">QTLSorghum</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#installation">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#examples">Examples:</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#load-install-libraries">Load/install libraries</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#set-the-working-directory-to-where-vcf-file-is-stored-in-file-system">Set the Working Directory to where VCF file is stored in file system</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#vcf-file-must-only-contain-bialleleic-variants-filter-upstream-e-g-with-bcftools-view-m2-m2-also-the-qtlseqr-functions-will-only-take-snps-ie-length-of-ref-and-alt-1">Vcf file must only contain bialleleic variants. (filter upstream, e.g., with bcftools view -m2 -M2), also the QTLseqR functions will only take SNPS, ie, length of REF and ALT== 1</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#call-the-parser">Call the Parser</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#preview-the-csv-file">Preview the CSV file</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#invoke-unique-command-to-extract-sample-names-reverse-comapatible-to-the-vcf">Invoke unique command to extract Sample names reverse comapatible to the VCF</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#inspect-the-head-of-the-df-object">Inspect the head of the df object</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#g-distribution-plot">G’ Distribution Plot</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#plot-g-statistic-distribution">Plot G Statistic Distribution</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#use-rmvp-package-to-view-snps-on-chromosomes-contigs">Use RMVP package to view SNPs on chromosomes/contigs</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#export-summary-csv">Export summary CSV</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#preview-the-summary-qtl">Preview the Summary QTL</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#p2-p1-qtl-is-present">p2 &gt;&gt; p1 QTL is present</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#however-ns-c-1-is-not-true">However, ns &gt;&gt; C &gt;&gt; 1 is NOT TRUE</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#standard-parser-output-makes-a-list-of-chromosome-as-named-in-vcf-file">Standard Parser Output makes a list of Chromosome as named in VCF file</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#also-find-unique-sample-names-reverse-compatible-with-vcf-file">Also find unique sample names reverse compatible with VCF file</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#id1">Preview the CSV file</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#plot-histograms-associated-with-filtering-arguments-to-determine-if-cut-off-values-are-appropriate">Plot histograms associated with filtering arguments to determine if cut off values are appropriate</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#id2">Plot G Statistic Distribution</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#lets-take-a-look-at-snps-per-chromosome-using-rmvp-package">Lets take a look at SNPs per chromosome using rMVP Package</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#id3">Export summary CSV</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#preview-the-qtl-summary">Preview the QTL Summary</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#looks-dense">Looks Dense</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#filter-low-allelic-depth-frequencies">Filter Low Allelic Depth Frequencies</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#preview-the-plot-with-idenitfied-snp-positions">Preview the plot with idenitfied SNP positions</a></li>
<li class="toctree-l1"><a class="reference internal" href="qtl.html#investigate-snp-pos-24525659">Investigate SNP&#64;POS 24525659</a></li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="#">Documentation overview</a><ul>
      <li>Next: <a href="qtl.html" title="next chapter">QTLSorghum</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2022, Michael Hall.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 4.4.0</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="_sources/index.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>